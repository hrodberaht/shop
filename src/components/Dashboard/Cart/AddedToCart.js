import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class AddedToCart extends Component {
  render() {
    const {
      product: {
        name, price, pcsOrder, totalPrice,
      },
      show,
      redirect,
    } = this.props;
    return (
      <div className="cart-popup">
        <div className="cart-popup__inside">
          <h4>Product added to cart:</h4>
          <p>
            <span>Name:</span>
            {name}
          </p>
          <p>
            <span>Price:</span>
            {price}
          </p>
          <p>
            <span>Pcs:</span>
            {pcsOrder}
          </p>
          <p>
            <span>Tota price:</span>
            {totalPrice}
          </p>
          <div className="cart-popup__buttons">
            <button id="more" className="btn btn-primary" type="button" onClick={show}>
              More
            </button>
            <button id="cart" className="btn btn-primary" type="button" onClick={redirect}>
              Cart
            </button>
          </div>
        </div>
      </div>
    );
  }
}

AddedToCart.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string,
    price: PropTypes.number,
    pcsOrder: PropTypes.number,
    totalPrice: PropTypes.number,
  }).isRequired,
  show: PropTypes.func.isRequired,
  redirect: PropTypes.func.isRequired,
};
