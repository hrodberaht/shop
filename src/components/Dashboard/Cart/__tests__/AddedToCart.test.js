import AddedToCart from '../AddedToCart';

const defaultProps = {
  product: {
    id: '1',
    name: 'fx',
    price: 400,
    pcsOrder: 2,
    totalPrice: 800,
  },
  show: jest.fn(),
  redirect: jest.fn(),
};

const setup = buildSetup(AddedToCart, defaultProps);

describe('<AddedToCart />', () => {
  it('should render witout crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call show after click', () => {
    const { wrapper } = setup();
    wrapper.find('#more').simulate('click');

    expect(wrapper.instance().props.show).toHaveBeenCalled();
  });

  it('should call redirect after click', () => {
    const { wrapper } = setup();
    wrapper.find('#cart').simulate('click');

    expect(wrapper.instance().props.redirect).toHaveBeenCalled();
  });
});
