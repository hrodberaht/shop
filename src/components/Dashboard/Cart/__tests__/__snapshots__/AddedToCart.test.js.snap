// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddedToCart /> should render witout crash 1`] = `
<div
  className="cart-background"
>
  <div
    className="cart-popup"
  >
    <div
      className="cart-popup__inside"
    >
      <h4>
        Product added to cart:
      </h4>
      <p>
        <span>
          Name:
        </span>
        fx
      </p>
      <p>
        <span>
          Price:
        </span>
        400
      </p>
      <p>
        <span>
          Pcs:
        </span>
        2
      </p>
      <p>
        <span>
          Tota price:
        </span>
        800.00
      </p>
      <div
        className="cart-popup__buttons"
      >
        <button
          className="btn btn-primary"
          id="more"
          onClick={[MockFunction]}
          type="button"
        >
          More
        </button>
        <button
          className="btn btn-primary"
          id="cart"
          onClick={[MockFunction]}
          type="button"
        >
          Cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
